### PR TITLE
Adding 3DI D300 v2

### PR DESCRIPTION
**Pull request by v27jain**
Hi,
We are a 3D printer manufacturer based in India (3Deometry Innovations - www.3deometry.com) and would like to add our machine 3DI D300 v2 to the Cura ecosystem.

### CODE REVIEW

**Comment by fieldOfView** [comment]
The file must have an extension `.def.json`

**Comment by mebgizmo** [comment]
I'm a person who does 3d print. I got a machine profile from sunlu for my sunlu S8 and would like to have it add to Cura
[Sunlu.fff File Information.txt](https://github.com/Ultimaker/Cura/files/4307137/Sunlu.fff.File.Information.txt)
[Sunlu S3D profile.zip](https://github.com/Ultimaker/Cura/files/4307139/Sunlu.S3D.profile.zip)


 please

**Comment by Ghostkeeper** [comment]
I think that's a profile for Simplify3D, Mebgizmo. It doesn't translate 1:1 to a Cura printer definition.

File: resources/definitions/3DI_D300_v2 [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{
```
Comment (Ghostkeeper) [inline resources/definitions/3DI_D300_v2]: This file has no file extension.

Comment (Ghostkeeper) [inline resources/definitions/3DI_D300_v2]: Please hold similar file naming conventions to the rest of the printers: lowercase with underscores.

Comment (Ghostkeeper) [inline resources/definitions/3DI_D300_v2]: I can't find any "v2" of this printer. Is it just v2 of the profile, or is there a specification page about this printer online?

File: resources/definitions/3DI_D300_v2 [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        "has_materials": true,
+        "machine_extruder_trains":
+        {
+            "0": "3DI_D300_v2_extruder_0"
+        }
+    },
+
+    "overrides": {
+        "machine_center_is_zero": { "default_value": true },
+        "machine_depth": { "default_value": 310 },
```
Comment (Ghostkeeper) [inline resources/definitions/3DI_D300_v2]: The specification, as far as I could find, specify that this printer is 300x300x300: https://3deometry.com/3d-printers/3deometry/3di-d300-3d-printer/

This is probably why it is called "D300" as well.

File: resources/definitions/3DI_D300_v2 [hunk 3]
```diff
@@ -0,0 +1,34 @@
+{
+    "version": 2,
+    "name": "3DI D300 v2",
+    "inherits": "fdmprinter",
+    "metadata": {
+        "visible": true,
+        "author": "V27Jain",
+        "manufacturer": "3Deometry Innovations",
+        "file_formats": "text/x-gcode",
+        "platform": "3DI_D300_v2_platform.STL",
```
Comment (Ghostkeeper) [inline resources/definitions/3DI_D300_v2]: This platform is not aligned to the build volume. It needs to be centred.
image

**Comment by nallath** [comment]
Closing this due to inactivity. Changes have been requested but they have not been resolved.

**Comment by v27jain** [comment]
Hi,
Wasn't feasible for me to work in these recent months. Can it be reopened
later? Or do I need to pull it and rework on it again?

Regards
Vaibhav Jain

**Comment by v27jain** [comment]
Thanks.

**Comment by nallath** [comment]
Closing this due to inactivity.

File: resources/definitions/3DI_D300_v2 (hunk 1 above)
Comment (v27jain) [reply resources/definitions/3DI_D300_v2:1]: The version was stopped. Right now we are making the version 2 while adding this to Cura

Comment (Ghostkeeper) [reply resources/definitions/3DI_D300_v2:1]: You can open a new PR once the requested changes are made, then.

**Pull request closed without merging**